src/training_hub/algorithms/lora_grpo.py

```
@@ -0,0 +1,1570 @@
+            if tasks is not None:
+                train_data = tasks
+            else:
+                train_data, _ = _load_generic_dataset(data_path, n_train)
+            val_data = []
+            mode = "custom"
+        elif data_path is not None and reward_fn is not None:
+            # Generic data with custom reward — auto-generate rollout
+            train_data, val_data = _load_generic_dataset(data_path, n_train)
+            mode = "generic"
+        elif data_path is not None:
+            train_data, val_data = _load_tool_call_dataset(
```

Ruff (0.15.11)

[warning] Unpacked variable `val_data` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)